tools/validators/instance_validator/error_guide.md

```diff
@@ -0,0 +1,1712 @@
+  cloud_device_id: "bad_device_id"
+  translation:
+    shade_extent_percentage_command:
+      present_value: "points.shade_extent_percentage_command.present_value"
+      units:
+        key: "points.shade_extent_percentage_command.units"
+        values:
+          percent: "%"
+
+US-SEA-BLDG1-GUID:
+  type: FACILITIES/BUILDING
+  code: US-SEA-BLDG1
+```
+
+### Error Message:
+```
+[ERROR]	Entity SDC_EXT-17-GUID (SDC_EXT-17) invalid cloud_device_id, please refer to the documentation: https://github.com/google/digitalbuildings/blob/master/ontology/docs/building_config.md#identifiers [0-9]{16}
+```
+
+## Entity Update Requirements
+
+### Description:
+When updating an existing entity, the etag and update_mask must be specified.
+
+### Example Bad Config:
+```
+CONFIG_METADATA:
+  operation: UPDATE
+
+US-SEA-BLDG1-GUID:
+  operation: UPDATE
+  type: FACILITIES/BUILDING
+  code: US-SEA-BLDG1
+```
+
+### Error Message:
+```
+while parsing a mapping
+  in "<unicode string>", line 1, column 1:
+    US-SEA-BLDG1-GUID:
+     ^ (line: 1)
+required key(s) 'etag', 'update_mask' not found
+  in "<unicode string>", line 4, column 1:
+      code: US-SEA-BLDG1
+    ^ (line: 4)
+```
+
+## Entity Export Requirements
+
+### Description:
+An etag is required on entities when performing an export operation.
+
+### Example Bad Config:
+```
+CONFIG_METADATA:
+  operation: UPDATE
+
+SDC_EXT-19-GUID:
+  type: HVAC/SDC_EXT
+  code: SDC_EXT-19
+  cloud_device_id: "baz"
+  operation: EXPORT
+```
+
+### Error Message:
+```
+while parsing a mapping
+  in "<unicode string>", line 1, column 1:
+    SDC_EXT-19-GUID:
+     ^ (line: 1)
+required key(s) 'etag' not found
+  in "<unicode string>", line 5, column 1:
+      operation: EXPORT
+    ^ (line: 5)
+```
+
+## Duplicate Cloud Device IDs
+
+### Description:
+Each entity must have a unique cloud device id.
+
+### Example Bad Config:
+```
+SDC_EXT-17-GUID:
+  type: HVAC/SDC_EXT
+  code: SDC_EXT-17
+  cloud_device_id: "2619178366980754"
+  translation:
+    shade_extent_percentage_command:
+      present_value: "points.shade_extent_percentage_command.present_value"
+      units:
+        key: "pointset.points.shade_extent_percentage_command.units"
+        values:
+          percent: "%"
+
+SDC_EXT-18-GUID:
+  type: HVAC/SDC_EXT
+  code: SDC_EXT-18
+  cloud_device_id: "2619178366980754"
+  translation:
+    shade_extent_percentage_command:
+      present_value: "points.shade_extent_percentage_command.present_value"
+      units:
+        key: "pointset.points.shade_extent_percentage_command.units"
+        values:
+          percent: "%"
+
+US-SEA-BLDG1-GUID:
+  type: FACILITIES/BUILDING
+  code: US-SEA-BLDG1
+```
+
+### Error Message:
+```
+[ERROR]	Duplicate cloud device id 2619178366980754 used for multiple entities:
+SDC_EXT-17-GUID: SDC_EXT-17
+SDC_EXT-18-GUID: SDC_EXT-18
+```
+
+## Too Many Operations
+
+### Description:
+A building config can only contain two operations at once, one of which being export.
```

Comment:
Please re-word to make more clear -- the config can only contain entities with two types of operations (clarify that there can still only be one CONFIG_METADATA operation defined)